src/guides/v2.3/extension-dev-guide/api-concepts.md

```diff
@@ -15,10 +15,84 @@ Public interfaces for a [module](https://glossary.magento.com/module) are marked
  {:.bs-callout-info}
 Third-party developers should use only these interfaces, that is, interfaces with the `@api` annotation. You can use other interfaces but those may be modified or removed in subsequent Magento releases. For more information, see [Backward compatibility]({{ page.baseurl }}/contributor-guide/backward-compatible-development/).
 
+### Example of public interface annotation
+
+```php
+namespace Magento\CatalogRule\Api;
+
+/**
+ * Interface CatalogRuleRepositoryInterface
+ * @api
+ * @since 100.1.0
+ */
+interface CatalogRuleRepositoryInterface
+{
+...
+```
+
 ## What is an API? {#api-definition}
 
 An application programming interface (API) is a set of interfaces and their implementations that a module provides to other modules.
 
+### Example of implementation a set of API interfaces
```

Comment:
Hi @DenisSaltanahmedov. I would suggest shrinking this example to just one interface declaration and the corresponding `di.xml` file. Also, I would add extra information about an interface implementation. Like "a default implementation for the interface may be declared in the following way ... ".  

So, just a little more information on what is the examples about. 

Thank you!